js/service.js

```diff
@@ -111,10 +111,49 @@ function callURL(route, params = {}) {
   });
 }
 
+/**
+ * Make an POST XHR request and return a promise to resolve it.
+ * @param  {Object} params A key-value object to be posted
+ * @returns {Promise} A promise to resolve an XHR request
+ */
+function postEntry(entryData) {
+  let request = new XMLHttpRequest();
+
+  return new Promise((resolve, reject) => {
+    request.open(`POST`, `${pulseAPI}/entries/`, true);
+
+    request.withCredentials = true;
+    request.onload = (event) => {
+      let result = event.currentTarget;
+
+      if (result.status >= 200 && result.status < 400) {
+        try {
+          resolve(JSON.parse(result.response));
```

Comment:
technically this could also fire when `resolve` throws, so a better solution is:
```
let data;

try {
  data = JSON.parse(result.response);
} catch (error) {
  // this error can only have come from JSON parsing
  reject(error);
}

resolve(data);
```